projects/vision/decode_jpeg_fuzz.cc

```diff
@@ -0,0 +1,79 @@
+// Copyright 2022 ISP RAS
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+//###############################################################################
+
+#include "decode_jpeg.h"
+#include <unistd.h>
+
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size) {
+  if (size <= 2) { return 0; }
+  size_t size_i = size - 1;
+  char name[] = "/tmp/torch-fuzz-XXXXXX";
+  char *dir = mktemp(name);
+  std::ofstream fp;
+  fp.open(dir, std::ios::out | std::ios::binary);
+  fp.write((char *)(data + 1), size_i);
+  fp.close();
+
+  if (size <= 0) {
+    unlink(dir);
+    return 0;
+  }
+
+  auto input_data =
+      torch::from_file(dir, /*shared=*/false, /*size=*/size_i, torch::kU8);
```

Comment:
Is it possible to create tensor from memory?
https://github.com/pytorch/pytorch/blob/a7edf713608806f10e17eab90d0a5df727d9a16e/aten/src/ATen/native/TensorFactories.cpp#L1463